lms/static/scripts/frontend_apps/components/JSTORPicker.tsx

```diff
@@ -126,6 +126,11 @@ export default function JSTORPicker({
     }
   };
 
+  const resetCurrentArticle = () => {
+    previousURL.current = null;
```

Comment:
Setting the reference to `null` makes the next call to `onURLChange` to evaluate `url && url === previousURL.current` as false.

We don't necessarily need to set `null`, any falsy value would work (like `''`), but I think `null` is easier to reason about, and has a better "match" with `setArticleId(null)`

Reply:
Alternatively, we could consider removing the `previousURL` reference entirely. My understanding is that it is used to prevent fetching the same content multiple times if `Enter` is pressed more than once without changing the content.

I'm not sure if this optimization is worth the extra complexity. Maybe something we can discuss. Or maybe there's another purpose which I'm missing.

Reply:
> My understanding is that it is used to prevent fetching the same content multiple times if Enter is pressed more than once without changing the URL.

The normal way of using this form with the keyboard is to enter a URL, press "Enter" once to search for the article, and "Enter" again to confirm the selection. It would be redundant to fetch the article details when Enter is pressed a second time, especially since it can take a couple of seconds to fetch metadata from JSTOR in the worst case.

> Alternatively, we could consider removing the previousURL reference entirely.

I realized the description of `previousURL` is slightly unclear about what it represents (what does "confirmed" mean?) and what it does represent is not really the state we want. In the HTML specs, the term used to describe what happens when an input field emits a "change" event, is that the value is ["committed"](https://html.spec.whatwg.org/multipage/indices.html#event-change). What this value represents is the last-committed value of the URL field.

What we want to know when the Enter key is pressed is "does the input field have uncommitted changes which are not reflected in `articleId`"? We could model this with a boolean value which is set to `true` when an `input` event occurs in the form and reset it when a `change` event happens.

Reply:
Oooh, I see. So I was indeed missing something.

Considering this, it does make sense to track the reference, either in the way it is now, or the alternative approach you suggested.

Thanks for the explanation 🙂